### PR TITLE
remote-install: add "ana/" to the default install list

### PR DESCRIPTION
**Pull request by oliver-sanders**
* The `ana/` directory is used by `rose_ana` to load "compotators" (custom comparison thinggies).
* These are typically run where the data is generated which is often remote.
* These directories typically contain a small number of Python files.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at https://github.com/cylc/cylc-doc/pull/535
- [ ] If this is a bug fix, PRs raised to both master and the relevant maintenance branch.

### CODE REVIEW

**Review by datamel — APPROVED** [review]

**Comment by datamel** [comment]
Just CHANGES.md conflicts.

**Comment by datamel** [comment]
Actually, while looking at the sibling docs pr I remembered that there is a bit of documentation in cylc-flow. In https://github.com/cylc/cylc-flow/blob/83f258c85ef6351de3e84cc0e5cfb6c5aa85d318/cylc/flow/cfgspec/workflow.py#L223, you could add `ana/`, to the list of dirs included by default.

**Comment by hjoliver** [comment]
> compotators

Is that a word?! :exploding_head:

**Comment by datamel** [comment]
> * compotators

https://www.merriam-webster.com/dictionary/compotator 😄

**Comment by oliver-sanders** [comment]
(I've called them "analysis modules" in the docs)

File: cylc/flow/cfgspec/workflow.py [hunk 1]
```diff
@@ -222,10 +222,17 @@ def get_script_common_text(this: str, example: Optional[str] = None):
...
-               * ``app/``
-               * ``bin/``
-               * ``etc/``
-               * ``lib/``
+               ``ana/``
+                  Rose ana analysis modules
+               ``app/``
+                  Rose applications
+               ``bin/``
+                  Cylc bin directory (added to PATH)
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/workflow.py:230]:
```suggestion
                  Cylc bin directory (added to ``PATH``)
```

File: cylc/flow/cfgspec/workflow.py [hunk 2]
```diff
@@ -222,10 +222,17 @@ def get_script_common_text(this: str, example: Optional[str] = None):
...
+               ``etc/``
+                  Miscellaneous resources
+               ``lib/``
+                  Cylc lib directory (lib/python added to PYTHONPATH for
+                  workflow config)
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/workflow.py:235]:
```suggestion
                  Cylc lib directory (``lib/python`` added to ``PYTHONPATH``
                  for workflow config)
```

**Review by MetRonnie — APPROVED** [review]
👍 Just a couple of optional formatting niceties suggested

**Comment by MetRonnie** [comment]
(fast test failures due to codecov upload fail)

**Pull request merged**